vault/provider.go

```diff
@@ -705,6 +706,16 @@ func providerToken(d *schema.ResourceData) (string, error) {
 	return strings.TrimSpace(token), nil
 }
 
+func RecordMultipleStates(lastStates ...string) func(*api.Response) {
+	return func(resp *api.Response) {
+		var m sync.Mutex
```

Comment:
I think we need to move this variable declaration outside the anonymous function, otherwise all invocations of the function will have their own mutex variable and not actually synchronise.

Reply:
Good call! Updated and moved the declaration outside the anonymous function but still under `RecordMultipleStates`